src/panels/lovelace/views/hui-view.ts

```diff
@@ -351,6 +455,63 @@ export class HUIView extends LitElement {
       curBadgeEl === badgeElToReplace ? newBadgeEl : curBadgeEl
     );
   }
+
+  private _storeLayout(): void {
+    let maxItems = 0;
+    const layout: LovelaceCardConfig[] = [];
+    const views: LovelaceViewConfig[] = [];
+
+    // Find max items in a column
+    this._grids.forEach((grid) => {
+      maxItems =
+        grid.getItems().length > maxItems ? grid.getItems().length : maxItems;
+    });
+
+    // Go through each column getting the 1st element from each and then the second element from each and so on
+    for (let i = 0; i < maxItems; i++) {
+      this._grids.forEach((grid) => {
+        if (!grid.getItems(i)[0]) {
+          return;
+        }
+
+        layout.push(grid.getItems(i)[0].getElement().cardConfig);
+      });
+    }
+
+    // Save Lovelace config - Yes I know - This isnt great
+    this.lovelace!.config.views.forEach((viewConf, index) => {
```

Comment:
Only save the card that was moved?

Reply:
I am not sure that it will be possible to only save the card that was moved. Because the cards are laid out 1 card per column then wrap to the first column but when the drag and drop happens the cards are in columns and shift up and down. Changing the position of one card will change many cards position.